synapse/push/mailer.py

```diff
@@ -317,9 +317,14 @@ async def send_email(self, email_address, subject, extra_template_vars):
     async def get_room_vars(
         self, room_id, user_id, notifs, notif_events, room_state_ids
     ):
-        my_member_event_id = room_state_ids[("m.room.member", user_id)]
-        my_member_event = await self.store.get_event(my_member_event_id)
-        is_invite = my_member_event.content["membership"] == "invite"
+        # Check if one of the notifs is an invite event for the user.
+        is_invite = False
+        for n in notifs:
+            ev = notif_events[n["event_id"]]
+            if ev.type == EventTypes.Member and ev.state_key == user_id:
+                if ev.content.get("membership") == Membership.INVITE:
+                    is_invite = True
+                    break
```

Comment:
This looks OK, I think, but would it be more efficient to still just try to grab the membership event directly?

```python
my_member_event_id = room_state_ids.get((EventTypes.Member, user_id))
is_invite = my_member_event_id and notif_events[my_member_event_id].content.get("membership") == Membership.INVITE
```

I'm not really sure if that makes sense ^ since I'm having trouble following what the inputs to this method are.

Reply:
The problem is really that `room_state_ids` may be empty if the server is no longer in the room (or has never been, in the case of remote invites, if they worked))

Reply:
Ah, OK. Thanks for explaining! I think it looks good. 👍